### ISSUE TITLE
Const typeguard flows into methods but not properties

### ISSUE THREAD

**Issue by jods4** [microsoft/TypeScript#45006]
# Bug Report

### 🔎 Search Terms

type guard get property const

### 🕗 Version & Regression Information

Fails at least on 4.3.5 and 4.4.0-beta.

- I was unable to test this on prior versions because I noticed this behavior very recently.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.4.0-beta#code/MYewdgzgLgBAHgLhtATgSzAcxgHxmAVwFsAjAUxRgF4YBZAQygAsA6FesAExCIAoBKGAB4YABhYBWGAH4YAIlRyYSAIwAmAMwBuALAAoNADMYvKAE8ADmRDG41KjQVR0WOYIDe+mDAA2ZWCAkAFbUMJ563t6Y-jAWKCAWAmEwKP4EKGDwLFAgAKoWVigAwvQQZEkAvgA0XpFE-kwgnEnuKWkZWTn5hSVllbUV+hVAA)

### 💻 Code

```ts
const x: string | number = Math.random() < 0.5 ? "str" : 123;
if (typeof x === "string") {
  let obj = {
    get prop() { return x.toUpperCase() },  // Error: property toUpperCase doesn't exist on number
    method() { return x.toUpperCase() } // Ok
  }
}
```

### 🙁 Actual behavior
The method compiles, the getter raises an error.

### 🙂 Expected behavior
Behaviour should be the same for both.
Given that `x` is a const and its type was narrowed to `string`, no error seems right.

**Comment by softwareCobbler** [comment]
should control flow into setters, too?

**Comment by jods4** [comment]
@softwareCobbler Yes.
```ts
const x: string | null = "abc";
if (x) {
 const obj = {
    // Following is valid and should not warn that x might be null
    set test(val) { console.log(x.toUpperCase(); }
  }
}
```
